File: order-service/Orders.Api/Models/OrderDetailsResponse.cs
using Orders.Api.Entities;
namespace Orders.Api.Models;

public class OrderDetailsResponse
{
    public Order Order { get; set; } = default!;
    public string? CustomerName { get; set; }
    public List<OrderLineDetails> OrderLines { get; set; } = new List<OrderLineDetails>();
}

public class OrderLineDetails
{
    public int Id { get; set; }
    public int ProductId { get; set; }
    public string ProductName { get; set; } = default!;
    public int Quantity { get; set; }
    public decimal Price { get; set; }
    public decimal Total { get; set; }
}

File: web/src/api/models/order/OrderDetailsResponse.ts
import type { Address } from '../shared/Address';
import type { Order } from './Order';
import type { OrderLineDetails } from './OrderLineDetails';
export type OrderDetailsResponse = {
    order?: Order;
    customerName?: string | null;
    billingAddress?: Address | null;
    shippingAddress?: Address | null;
    orderLines?: Array<OrderLineDetails> | null;
};

